.eslintrc.js

```diff
@@ -1,96 +1,130 @@
+const parserOptions = {
+    ecmaVersion: 6,
+    sourceType: 'module',
+}
+
+const eslintRules = {
+    'arrow-body-style': 'error',
+    'arrow-spacing': 'error',
+    'eol-last': 'error',
+    'func-call-spacing': 'error',
+    indent: 'off', // OFF b/c causes problems between Prettier and ESLint
+    'linebreak-style': 'off', // OFF b/c Windows (Git) puts CRLF line endings
+    'missing-declaration': 'off', // OFF b/c throws errors on imports / require statements
+    'multiline-ternary': 'off', // OFF b/c causes problems between Prettier and ESLint
+    'no-alert': 'error',
+    'no-async-promise-executor': 'off', // OFF b/c used in lib/migration.ts
+    'no-case-declarations': 'error',
+    'no-console': ['error', { allow: ['error', 'warn'] }],
+    'no-control-regex': 'error',
+    'no-dupe-keys': 'error',
+    'no-empty': 'error',
+    'no-extra-boolean-cast': 'error',
+    'no-extra-parens': 'off', // OFF b/c reactive dependencies in .svelte files (mostly for errors)
+    'no-extra-semi': 'error',
+    'no-fallthrough': 'error',
+    'no-import-assign': 'error',
+    'no-irregular-whitespace': 'off', // OFF b/c special char used in packages/desktop/electron/lib/keychain.js
+    'no-prototype-builtins': 'error',
+    'no-return-await': 'error',
+    // 'no-tab': ['error', { allowIndentationTabs: true }],
+    'no-trailing-spaces': 'error',
+    'no-useless-escape': 'error',
+    'no-undef': 'error',
+    'no-underscore-dangle': 'off', // OFF b/c this syntax is used for defining local callback methods
+    'no-unreachable': 'error',
+    'no-unused-export-let': 'off', // OFF b/c troublesome with some .js files in packages/shared
+    'no-unused-vars': 'off', // OFF b/c there are simply too many and they're harmless
+    'no-var': 'error',
+    'prefer-arrow-callback': 'warn',
+    'prefer-const': 'warn',
+    'prefer-destructuring': 'off', // OFF b/c it's not really correct
+    quotes: ['error', 'single'],
+    semi: 'off', // OFF b/c we aren't using semicolons
+    'space-before-function-paren': 'off', // OFF b/c we aren't using spaces before function parameters / signatures
+    'spaced-comment': 'off', // OFF b/c there are lots of comments with mixed formatting - best to leave it
+}
+
+const typescriptEslintRules = {
+    '@typescript-eslint/array-type': 'error',
+    '@typescript-eslint/await-thenable': 'error',
+    '@typescript-eslint/ban-types': 'error',
+    '@typescript-eslint/ban-ts-comment': 'off', // OFF b/c ts-ignore comments are sometimes useful and needed
+    '@typescript-eslint/explicit-module-boundary-types': 'error',
+    '@typescript-eslint/no-array-constructor': 'error',
+    '@typescript-eslint/no-empty-function': 'off', // OFF b/c we use empty functions a lot (esp. for initialization)
+    '@typescript-eslint/no-explicit-any': 'off', // OFF b/c we use any but only when it's necessary
```

Comment:
Yes I was conflicted about this one because I prefer strict typing. There's quite a lot of callback functions in the Svelte components that would be good for typing, but it's definitely needed for stuff in `shared/lib`. I am not sure how long it would take me to comb through those callback functions and add types to them.